API/Controllers/PlayersController.cs

```diff
@@ -30,31 +30,19 @@ public async Task<IActionResult> GetAllAsync()
         return Ok(players);
     }
 
-    [HttpGet("{osuId:long}/info")]
-    public async Task<ActionResult<PlayerInfoDTO?>> GetByUserIdAsync(long osuId)
+    [HttpGet("{key}/info")]
+    public async Task<ActionResult<PlayerInfoDTO?>> GetAsync(string key)
```

Comment:
```suggestion
    [EndpointSummary("Get player info by versatile search")]
    [EnpointDescription("Get player info searching first by id, then osuId, then username")]
    public async Task<ActionResult<PlayerInfoDTO?>> GetAsync(string key)
```

Don't forget your documentation >:)

Reply:
so huge